Simplenote/SPAppDelegate.m

```diff
@@ -683,6 +704,17 @@ - (void)bucketDidFinishIndexing:(SPBucket *)bucket
 {
     if ([bucket.name isEqualToString:@"Note"]) {
         [_noteListViewController setWaitingForIndex:NO];
+        
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
+            NSArray *deleted = [bucket objectsForPredicate:[NSPredicate predicateWithFormat:@"deleted == YES"]];
+            if (deleted.count > 0) {
+                NSArray *deletedIDs = [deleted valueForKey:@"simperiumKey"];
+                [[CSSearchableIndex defaultSearchableIndex] deleteSearchableItemsWithIdentifiers:deletedIDs completionHandler:nil];
+            }
+            
+            NSArray *notes = [bucket objectsForPredicate:[NSPredicate predicateWithFormat:@"deleted == NO"]];
+            [[CSSearchableIndex defaultSearchableIndex] indexSearchableNotes:notes];
+        });
```

Comment:
I was considering `didChangeObjectForKey` too, but it would fire on every object separately. I'll do that on a private context instead.

The `deleted` flag is actually a field of Note, used to differentiate notes in the thrash. The one you're thinking about is `isDeleted` and I completely agree with the rest of the comment.